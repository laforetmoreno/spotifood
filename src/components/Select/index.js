import React from 'react';
import { func, string, arrayOf, bool, object, oneOfType, number } from 'prop-types';

import styles from './index.module.scss';

const Select = ({ onChange, placeholder, disabled, options, className, name, label }) => {
  const handleChange = e => {
    if (onChange) onChange(e.target.value, e.target.name);
  };

  return (
    <div className={styles.wrapper}>
      <span className={styles.label}>{label}</span>
      <select
        disabled={disabled}
        className={`${styles.select} ${className} ${disabled && styles.disabled}`}
        onChange={handleChange}
        name={name}
        defaultValue="default"
        data-testid="select"
      >
        <>
          <option value="default">{placeholder}</option>
          {options.map(option => (
            <option
              key={option.value ? option.value : option}
              value={option.value ? option.value : option}
            >
              {option.name ? option.name : option}
            </option>
          ))}
        </>
      </select>
    </div>
  );
};

Select.propTypes = {
  onChange: func,
  options: arrayOf(oneOfType([string, object])),
  placeholder: oneOfType([number, string]),
  disabled: bool,
  className: string,
  name: string,
  label: string,
};

Select.defaultProps = {
  onChange: () => null,
  options: [],
  placeholder: '',
  disabled: false,
  className: '',
  name: '',
  label: '',
};

export default Select;
