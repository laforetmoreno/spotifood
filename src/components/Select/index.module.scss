@import '../../styles/';

.wrapper {
  @include flex-column();
  width: 100%;
}

.select {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  color: $--color-dark-blue;
  font-size: rem(14px);
  font-variant: tabular-nums;
  line-height: 1.5;
  list-style: none;
  position: relative;
  display: inline-block;
  outline: 0;
  padding: rem(5px);
  border-radius: rem(4px);
  background: $--color-green;
  border: rem(1px) solid transparent;
}

.disabled {
  opacity: 0.5;
}

.label {
  margin-left: rem(10px);
  color: $--color-green;
}
