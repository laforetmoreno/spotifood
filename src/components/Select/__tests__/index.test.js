import React from 'react';
import renderer from 'react-test-renderer';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import Select from '../index';

describe('<Select />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Select
          placeholder=""
          disabled={false}
          options={[{ name: 'fake name', value: 'fake value' }]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('get by id and change', async () => {
    const { getByTestId } = render(
      <Select
        placeholder=""
        disabled={false}
        options={[{ name: 'fake name', value: 'fake value' }]}
      />,
    );
    const fieldNode = await waitForElement(() => getByTestId('select'));
    fireEvent.change(fieldNode);
  });
});
