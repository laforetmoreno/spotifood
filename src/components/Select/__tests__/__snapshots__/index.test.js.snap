// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly 1`] = `
<div
  className="wrapper"
>
  <span
    className="label"
  >
    
  </span>
  <select
    className="select  false"
    data-testid="select"
    defaultValue="default"
    disabled={false}
    name=""
    onChange={[Function]}
  >
    <option
      value="default"
    >
      
    </option>
    <option
      value="fake value"
    >
      fake name
    </option>
  </select>
</div>
`;
